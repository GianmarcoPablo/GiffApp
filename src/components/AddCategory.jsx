import { useState } from "react";

const AddCategory = ({ obtenerCategoria }) => {

    const [inputValue, setInputValue] = useState("");


    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length <= 1)return
        obtenerCategoria(inputValue)
        setInputValue("")
       
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder="Buscar Gif"
                value={inputValue}
                onChange={(e) => setInputValue(e.target.value)}
            />
        </form>
    );
};

export default AddCategory;
