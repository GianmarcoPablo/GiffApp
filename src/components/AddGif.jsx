import { useState } from "react"

const AddGif = ({obtenerCategorias}) => {

    const [inputValue, setInputValue] = useState('')

    const handleSubmit = (e) =>{
        e.preventDefault()
        if(inputValue.trim().length < 1){
            alert("agrega algo")
        }else{
            obtenerCategorias(inputValue)
        }
        setInputValue("")   
    }
    return (
        <form
            onSubmit={handleSubmit}
        >
            <input 
                type="text"
                value={inputValue}
                onChange={(e)=>setInputValue(e.target.value)}
            />
        </form>
    )
}

export default AddGif