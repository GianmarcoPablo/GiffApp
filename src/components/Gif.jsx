import PropTypes from 'prop-types'

export const Gif = ({ title, url }) => {
    return (
        <div className="card">
            <img
                src={url}
                alt={`imagen de ${title}`}
            />
            <p>{title}</p>
        </div>
    )
}

//proptypes
Gif.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}