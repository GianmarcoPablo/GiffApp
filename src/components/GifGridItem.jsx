import React from 'react'

const GifGridItem = ({title,url}) => {
    return (
        <div className='card'>
            <img src={url} alt={title} />
            <p>{title}</p>
            <a
                href    ={url}
                target='_blank' 
            >
                enlace  
            </a>
        </div>
    )
}

export default GifGridItem