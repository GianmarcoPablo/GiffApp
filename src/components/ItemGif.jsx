import Gif from "./Gif"
import useFecht from "../hooks/useFecht"

const ItemGif = ({categoria}) => {

    const {gifs} = useFecht(categoria)

    return (
        <div className='card'>
            {categoria}
            <div className="card-grid">
                {gifs?.map((item)=>(
                    <Gif
                        key={item.id}
                        {...item}
                    />
                ))}
            </div>
        </div>
    )
}

export default ItemGif