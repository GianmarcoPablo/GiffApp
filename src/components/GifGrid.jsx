import React, { useEffect, useState } from "react";
import GifGridItem from "./GifGridItem";
import { useFechGifs } from "../hooks/useFechGifs";

const GifGrid = ({ category  }) => {

    const {images,isLoading} = useFechGifs(category);
    
    return (
        <div>
            <h3>{category}</h3>
            {isLoading && <p>Cargando...</p>}
            <div className="card-grid">
                {images.map((image) => (
                    <GifGridItem
                        key={image.id}
                        {...image}
                    />
                ))}
            </div>
        </div>
    );
};  

export default GifGrid;
