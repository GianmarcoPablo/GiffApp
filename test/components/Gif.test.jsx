import { Gif } from '../../src/components/Gif'
import {render,screen} from '@testing-library/react'

describe('Pruebas en GifItem', () => { 

    const title = 'Un titulo'
    const url = 'https://localhost/algo.jpg'

    test('debe de hacer match con el snapshot', () => { 
        const {container} = render(<Gif
            title={title}
            url={url}
        />)
        expect(container).toMatchSnapshot()
    })
    test('debe de mostrar la imagen on el url y alt indicado', () => { 
        render(<Gif title={title} url={url}/>)
        const {src,alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })
    test("debe de mostrar el titulo en el componente",()=>{ 
        render(<Gif title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()
    })
})