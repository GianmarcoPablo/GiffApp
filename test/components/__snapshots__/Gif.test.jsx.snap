// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="imagen de Un titulo"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
